src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/ConfinedScope.java

```diff
@@ -121,6 +121,7 @@ public void close() {
             if (!released) {
                 released = true;
                 lockCount--;
+                Reference.reachabilityFence(ConfinedScope.this);
```

Comment:
Not sure this is needed - I added it as we were worried that possible optimization (e.g. scalarization) might break reachability (this class is an innner class and keeps a strong reference to its enclosing, after all).